strawberryfields/engine.py

```diff
@@ -427,15 +428,15 @@ def run(self, program, *, args=None, compile_options=None, **kwargs):
         # pylint: disable=import-outside-toplevel
         from strawberryfields.tdm.tdmprogram import reshape_samples
 
-        # At this time we do not support lists of tdm programs
-        valid_tdm_program = (
-            not isinstance(program, collections.abc.Sequence) and program.type == "tdm"
-        )
-        if valid_tdm_program:
+        rolled_input = False
```

Comment:
I'm confused: what does rolled input mean? :thinking:

Reply:
`rolled_input` refers to whether the program that is passed is a rolled up `TDMProgram` (i.e. that `prog.circuit` is a rolled circuit with placeholders instead of values) or if it's already been unrolled by the user. Either way, it should be kept the same as it was before whenever everything has been run, and not change to rolled/unrolled.

Reply:
Thank you, that makes sense! I'm thinking that the name might be made a bit more descriptive to reflect what's going on (though now it makes sense). How about `received_as_rolled` or `received_rolled_circuit`?